src/algorithms/neal2_algorithm.h

```diff
@@ -53,7 +53,11 @@ class Neal2Algorithm : public MarginalAlgorithm {
   ~Neal2Algorithm() = default;
   Neal2Algorithm() = default;
 
-  std::string get_id() const override { return "Neal2"; }
+  bool requires_conjugate_hierarchy() { return true; }
```

Comment:
This should be a virtual, abstract method in the base class which is overridden in each derived class. Or maybe, even better, a virtual method in the base class with a certain default value (true?) which is overridden only as needed